src/pkg/blob/dao/dao.go

```diff
@@ -389,3 +392,19 @@ func (d *dao) DeleteBlob(ctx context.Context, id int64) error {
 	}
 	return nil
 }
+
+func (d *dao) GetBlobsNotRefedByProjectBlob(ctx context.Context, timeWindow int64) ([]*models.Blob, error) {
+	var noneRefed []*models.Blob
+	ormer, err := orm.FromContext(ctx)
+	if err != nil {
+		return noneRefed, err
+	}
+
+	sql := fmt.Sprintf(`SELECT b.* FROM blob AS b LEFT JOIN project_blob pb ON b.id = pb.blob_id WHERE pb.id IS NULL AND b.update_time <= now()- interval '%d hours';`, timeWindow)
```

Comment:
Do we need pagination here because the result may very large?

Reply:
will file an issue, and consider to add it in the next sprint.